lib/method_object.rb

```diff
@@ -69,7 +69,7 @@ def candidates_for_method_missing(method_name)
   end
 
   def define_delegated_method(delegate)
-    self.class.class_eval(
+    code =
       if delegate.method_to_call_on_delegate.to_s.end_with?('=')
```

Comment:
Holy how, why isn't this a guard clause?

Reply:
A guard clause may make sense for the purpose of extracting the code snippet to `eval`ed.  Is that what you are thinking? As it is seems fine since it is a 2 path conditional and is pretty easy to read.

Reply:
If the condition is false, then you end up doing `self.class.class_eval(nil)`.

Reply:
No, I'm wrong. I was looking at the diff not the raw code.